dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionResource.java

```diff
@@ -261,13 +263,7 @@ public Collection addEntryToCollection(@ApiParam(hidden = true) @Parameter(hidde
                 .withInitiatorUser(user)
                 .withType(Event.EventType.ADD_TO_COLLECTION);
 
-        if (entryAndCollection.getLeft() instanceof BioWorkflow) {
-            eventBuild = eventBuild.withBioWorkflow((BioWorkflow)entryAndCollection.getLeft());
-        } else if (entryAndCollection.getLeft() instanceof Service) {
-            eventBuild = eventBuild.withService((Service)entryAndCollection.getLeft());
-        } else if (entryAndCollection.getLeft() instanceof Tool) {
-            eventBuild = eventBuild.withTool((Tool)entryAndCollection.getLeft());
-        }
+        eventBuild = modifyBuilderForCollection(entryAndCollection, eventBuild);
```

Comment:
I think you made this same comment on an earlier PR of mine, but you can instead have on lines 258/260 and 316/312

```
Event.Builder eventBuild = entryAndCollection.getLeft().getEventBuilder()
  .withOrganization(...
```

Reply:
past me had good tips